src/prefect/engine.py

```diff
@@ -253,27 +267,51 @@ async def retrieve_flow_then_begin_flow_run(
         )
         return state
 
+    try:
+        parameters = get_call_parameters(
+            flow.fn, call_args=(), call_kwargs=flow_run.parameters
+        )
+    except Exception as exc:
+        message = f"Encountered an unexpected error when retrieving flow run parameters: {exc}"
+        flow_run_logger(flow_run).exception(message)
+        state = Failed(message=message, data=safe_encode_exception(exc))
+        await client.set_flow_run_state(
+            state=state, flow_run_id=flow_run_id, force=True
+        )
+        return state
+
     await client.update_flow_run(
         flow_run_id=flow_run_id,
         flow_version=flow.version,
+        parameters=parameters,
     )
 
     if flow.should_validate_parameters:
+        failed_state = None
         try:
-            parameters = flow.validate_parameters(flow_run.parameters)
+            parameters = flow.validate_parameters(parameters)
+        except (ParameterTypeError, SignatureMismatchError) as exc:
+            flow_run_logger(flow_run).error(str(exc))
```

Comment:
```suggestion
            flow_run_logger(flow_run).exception("Flow parameter validation failed.")
```
using `.exception` will log the traceback 👍